src/Base/Body.php

```diff
@@ -165,6 +165,30 @@ protected function matchArray($name, $schema, $body, $type)
         return true;
     }
 
+    /**
+     * Checks if the value is valid date.
+     *
+     * @param $name
+     * @param $schema
+     * @param $body
+     * @param $type
+     *
+     * @return bool|null
+     * @throws NotMatchedException
+     */
+    protected function matchDate($name, $schema, $body, $type)
+    {
+        if ($type !== 'date') {
+            return null;
+        }
+
+        if (!(bool)strtotime($body)) {
```

Comment:
Use `strtotime($body) === false`. The integer number zero is a non-error return value as per https://www.php.net/manual/en/function.strtotime.

That said, PHP does funny things with dates, like even parsing `yesterday 12:00`. I'm not sure about `strtotime()`, but I don't think this happens to match the OpenAPI standard by chance. What is the delta between this and the standard?

Reply:
Ok, I've checked the specification and I've got it wrong - the `date` should not be implemented.

I should use 

`An optional format modifier serves as a hint at the contents and format of the string. OpenAPI defines the following built-in string formats:
date – full-date notation as defined by RFC 3339, section 5.6, for example, 2017-07-21`

At this moment `matchString` does not validate anything -- we could probably add a support for it?

Reply:
So I've prepare format validation, but this could do a breaking change (tests fails because date-time has date string). 

1. We can remove the matchDate and format validation and leave the PR only for custom request + trait
2. We can add the date format + future validations